react/class_components/component_lifecycle_methods.md

```diff
@@ -0,0 +1,75 @@
+
+We'll start with a short recap of what a components lifecycle is. There are three stages to a components life: mounting, updating, and unmounting. Each of these have a method assigned to them within class components, which is what we are going to cover in this lesson.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+*   How to use lifecycle methods in a class component
+
+### render()
+
+The render function is the most used lifecycle method, and one that you've come across in the last class components lesson. It is the only required lifecycle method in a class component. It runs on mount and update of a component. Render should be pure, meaning it doesn't modify component state, returns the same thing each time it's called (given the same inputs), and doesn't directly interact with the browser. 
+
+### componentDidMount()
+
+This method is run after the component is mounted (inserted in the DOM tree). You should make any calls to fetch data that is needed for the component here. It is also a good place to do anything that is reliant on the component, such as drawing on a canvas element that you just rendered.
+
+### componentDidUpdate()
+
+This method is run after a component re-renders. Because of this, you have to be careful about what you update in this method, as if you're updating state indiscriminately, a re-render is caused, and you'll end up in an endless loop. You can avoid this issue by using conditional statements about the equality of previous and current props when updating state.
+
+In this method you should be updating anything that needs to be changed in response to either the DOM changing, or any states that you might want to act on upon change. For example, you'd want to refetch user data if the user changes.
+
+### componentWillUnmount()
+
+This is the last lifecycle method, which is called before a component is unmounted and destroyed. In this method you should be performing cleanup actions, so that would be cancelling network requests, clearing timers, etc.
+
+### useEffect()
+
+Now that we've learnt about class lifecycle methods, it's useful to understand that the `useEffect` hook used in functional components is essentially a combination of componentDidMount, componentDidUpdate, and componentWillUnmount. Which method/methods it relates to varies based on it's dependency array, and if it returns anything.
+
+*   An empty dependency array would be equivalent to `componentDidMount`.
+*   A dependency array with a value/values in it would be similar to `componentDidUpdate`, but only ran when the dependency provided changes.
+*   No dependency array would be equivalent to `componentDidMount` and `componentDidUpdate` combined.
+*   A return function inside of a `useEffect()` hook would be equivalent to `componentWillUnmount`.
+
+For example:
+
+~~~js
+    useEffect(() => {
+        placeholderFunction()
+        return () => clearUpFunction()
```

Comment:
Can we call it a `cleanupFunction` instead of `clearUpFunction`? I've seen the term `cleanup` used more often. I'm not tied to this suggestion; I'll leave the decision to you.

```suggestion
        return () => cleanupFunction()
```

Reply:
Yeah that sounds like a good change, I typically use the term cleanup more often too, not sure why I opted for clearup here.